### PR TITLE
Fix #812 Add Perf API functional tests

### LINKED ISSUE #812: Add functional tests for cFE Performance Monitor APIs
**Is your feature request related to a problem? Please describe.**
Need open source functional tests for certifiability

**Describe the solution you'd like**
Add functional tests for cFE Performance Monitor APIs
– CFE_ES_PerfLogEntry - Entry marker for use with Software Performance Analysis Tool.
– CFE_ES_PerfLogExit - Exit marker for use with Software Performance Analysis Tool.
– CFE_ES_PerfLogAdd - Function called by CFE_ES_PerfLogEntry and CFE_ES_PerfLogExit macros.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Requester Info**
Jacob Hageman - NASA/GSFC

### PR DESCRIPTION
**Pull request by pepepr08**
**Describe the contribution**
Fixes #812 
Add Perf API functional tests and Add Perf Trigger and Perf Start commands to functional test workflow.
Functional Test will now send a command to set a Perf trigger, send command to start capturing Perf data and finally send command to start running the functional tests.

**Testing performed**
Steps taken to test the contribution:
1. Build with tests enabled
2. Run cFS
3. Send commands to set Perf trigger, and start Perf data
4. Send command to run Functional Tests
5. Verify tests pass
6. Push code to Github and verify workflow works

**Expected behavior changes**
New tests added to functional tests. Workflow now sends two more commands to cFS.

**System(s) tested on**
 - OS: Ubuntu 18.04 VM

**Contributor Info - All information REQUIRED for consideration of pull request**
Jose F. Martinez Pedraza / NASA GSFC

### CODE REVIEW

**Comment by astrogeco** [comment]
[CCB:2021-08-11](https://github.com/nasa/cFS/wiki/Community-CCB-Agenda:-August-11-2021) APPROVED

File: .github/workflows/functional-tests.yml [hunk 1]
```diff
@@ -63,10 +63,13 @@ jobs:
       - name: List cpu1
         run: ls build/exe/cpu1/
 
+      # Run cFS, send commands to set perf trigger and start perf data, and run functional tests
       - name: Run cFS
         run: |
           ./core-cpu1 &
           sleep 10
+          ../host/cmdUtil --pktid=0x1806 --cmdcode=17 --endian=LE --uint32=3 --uint32=0x40000000 &
+          ../host/cmdUtil --pktid=0x1806 --cmdcode=14 --endian=LE --uint32=2 &
```
Comment (astrogeco) [inline .github/workflows/functional-tests.yml:72]: Remove `&` and bring commands to foreground to avoid race conditions

**Review by astrogeco — CHANGES_REQUESTED** [review]

**Comment by astrogeco** [comment]
[CCB:2021-08-11](https://github.com/nasa/cFS/wiki/Community-CCB-Agenda:-August-11-2021) APPROVED with CHANGES

File: .github/workflows/functional-tests.yml (hunk 1 above)
Comment (pepepr08) [reply .github/workflows/functional-tests.yml]: Done

File: modules/cfe_testcase/CMakeLists.txt [hunk 2]
```diff
@@ -11,6 +11,7 @@ add_cfe_app(cfe_testcase
     src/es_cds_test.c
     src/es_counter_test.c
     src/es_misc_test.c
+    src/es_perf_test.c
```
[the remark this answers is not in the document]
Comment (pepepr08) [reply modules/cfe_testcase/CMakeLists.txt]: Done

**Comment by pepepr08** [comment]
I split the test into four functions. I think log looks better now.
```
[BEGIN] 20 Test PerfLogEntry
[ PASS] 20.001 es_perf_test.c:36 - CFE_ES_PerfLogEntry(0)
[ PASS] 20.002 es_perf_test.c:37 - CFE_ES_PerfLogEntry(CFE_MISSION_ES_PERF_MAX_IDS - 1)
[ PASS] 20.003 es_perf_test.c:38 - CFE_ES_PerfLogEntry(CFE_MISSION_ES_PERF_MAX_IDS)
[  END] 20 Test PerfLogEntry    TOTAL::3     PASS::3     FAIL::0     MIR::0     TSF::0     TTF::0     N/A::0   

[BEGIN] 21 Test PerfLogExit
[ PASS] 21.001 es_perf_test.c:43 - CFE_ES_PerfLogExit(0)
[ PASS] 21.002 es_perf_test.c:44 - CFE_ES_PerfLogExit(CFE_MISSION_ES_PERF_MAX_IDS - 1)
[ PASS] 21.003 es_perf_test.c:45 - CFE_ES_PerfLogExit(CFE_MISSION_ES_PERF_MAX_IDS)
[  END] 21 Test PerfLogExit     TOTAL::3     PASS::3     FAIL::0     MIR::0     TSF::0     TTF::0     N/A::0   

[BEGIN] 22 Test PerfLogAdd
[ PASS] 22.001 es_perf_test.c:50 - CFE_ES_PerfLogAdd(0, 0)
[ PASS] 22.002 es_perf_test.c:51 - CFE_ES_PerfLogAdd(0, 1)
[ PASS] 22.003 es_perf_test.c:52 - CFE_ES_PerfLogAdd(0, 0xFFFFFFFF)
[ PASS] 22.004 es_perf_test.c:53 - CFE_ES_PerfLogAdd(CFE_MISSION_ES_PERF_MAX_IDS - 1, 0)
[ PASS] 22.005 es_perf_test.c:54 - CFE_ES_PerfLogAdd(CFE_MISSION_ES_PERF_MAX_IDS - 1, 1)
[ PASS] 22.006 es_perf_test.c:55 - CFE_ES_PerfLogAdd(CFE_MISSION_ES_PERF_MAX_IDS - 1, 0xFFFFFFFF)
[ PASS] 22.007 es_perf_test.c:56 - CFE_ES_PerfLogAdd(CFE_MISSION_ES_PERF_MAX_IDS, 0)
[ PASS] 22.008 es_perf_test.c:57 - CFE_ES_PerfLogAdd(CFE_MISSION_ES_PERF_MAX_IDS, 1)
[ PASS] 22.009 es_perf_test.c:58 - CFE_ES_PerfLogAdd(CFE_MISSION_ES_PERF_MAX_IDS, 0xFFFFFFFF)
[  END] 22 Test PerfLogAdd      TOTAL::9     PASS::9     FAIL::0     MIR::0     TSF::0     TTF::0     N/A::0   

[BEGIN] 23 Test Perf Trigger
[ PASS] 23.001 es_perf_test.c:64 - CFE_ES_PerfLogEntry(126)
[ PASS] 23.002 es_perf_test.c:65 - CFE_ES_PerfLogExit(126)
[  END] 23 Test Perf Trigger    TOTAL::2     PASS::2     FAIL::0     MIR::0     TSF::0     TTF::0     N/A::0   
```

**Review by zanzaben — APPROVED** [review]

**Comment by pepepr08** [comment]
Split single commit into two commits.

**Review by astrogeco — APPROVED** [review]

**Pull request merged**